### PR TITLE
Add eagerScalingStrategy for ScaledJob

### LINKED ISSUE #5114: Add Eager mode for ScaledJobs
### Proposal

KEDA doesn't scale up job number to the max according to[ this piece of description and information around](https://keda.sh/docs/2.11/concepts/scaling-jobs/#:~:text=Number%20of%20the%20Scale%3A%20The%20number%20of%20the%20job%20that%20is%20created.). But in our case, we want to launch as many jobs as possible and keep the queue as empty. So, can we have an Eager mode in addition so users like me can shorten the waiting time in general?

### Use-Case

We want to launch as many jobs as possible and keep the queue as empty.

### Is this a feature you are interested in implementing yourself?

Maybe

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by junekhan**
Added `eagerScalingStrategy` for `ScaledJob`

### Checklist

- [x] When introducing a new scaler, I agree with the [scaling governance policy](https://github.com/kedacore/governance/blob/main/SCALERS.md)
- [x] I have verified that my change is according to the [deprecations & breaking changes policy](https://github.com/kedacore/governance/blob/main/DEPRECATIONS.md)
- [x] Tests have been added
- [x] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)
- [x] A PR is opened to update our Helm chart ([repo](https://github.com/kedacore/charts)) *(if applicable, ie. when deployment manifests are modified)*
- [x] A PR is opened to update the documentation on ([repo](https://github.com/kedacore/keda-docs)) *(if applicable)*
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))

Fixes #5114 

Relates to #

### CODE REVIEW

**Review by zroubalik — COMMENTED** [review]
@junekhan thanks for the contribution, could you please also cover this scenario in [e2e tests](https://github.com/kedacore/keda/tree/main/tests)?

**Review by zroubalik — COMMENTED** [review]
Could you please open pr in https://github.com/kedacore/keda-docs/ to cover this new feature?

**Comment by junekhan** [comment]
> Could you please open pr in https://github.com/kedacore/keda-docs/ to cover this new feature?

Sure, thanks! Will do that.

**Comment by junekhan** [comment]
@zroubalik Hi there, I'm sorry that the e2e test didn't go well. I'm glad to analyze the logs, although it seems that I don't have access to it.

**Comment by zroubalik** [comment]
@junekhan you should be able to access the logs by following these steps (let me know if you are not):
Screenshot 2024-06-25 at 22 18 29
Screenshot 2024-06-25 at 22 17 34
Screenshot 2024-06-25 at 22 18 01

**Comment by junekhan** [comment]
@zroubalik Thanks for your input. I have no problem accessing the test suite log you mentioned, but I was looking for the log from inside the cluster, which can contain information about how the scaling decision was made. The test suite log only shows that the scaling didn't happen a single time, otherwise, I didn't find any clues.

**Comment by zroubalik** [comment]
> @zroubalik Thanks for your input. I have no problem accessing the test suite log you mentioned, but I was looking for the log from inside the cluster, which can contain information about how the scaling decision was made. The test suite log only shows that the scaling didn't happen a single time, otherwise, I didn't find any clues.

At the bottom of the output, there's log from KEDA operator as well. That's the best we can do, you can always run the failed test locally:  https://github.com/kedacore/keda/tree/main/tests#specific-test

**Comment by junekhan** [comment]
> > @zroubalik Thanks for your input. I have no problem accessing the test suite log you mentioned, but I was looking for the log from inside the cluster, which can contain information about how the scaling decision was made. The test suite log only shows that the scaling didn't happen a single time, otherwise, I didn't find any clues.
> 
> At the bottom of the output, there's log from KEDA operator as well. That's the best we can do, you can always run the failed test locally: https://github.com/kedacore/keda/tree/main/tests#specific-test

@zroubalik Thanks for your hint! That looks daunting 😨  but I will try.

**Comment by junekhan** [comment]
I got this error when setting up the e2e test. Any thoughts?

```
GOOS="darwin" go test -v -tags e2e ./utils/setup_test.go
# runtime/cgo
_cgo_export.c:3:10: fatal error: 'stdlib.h' file not found
FAIL	command-line-arguments [build failed]
FAIL
```

**Comment by junekhan** [comment]
> I got this error when setting up the e2e test. Any thoughts?
> 
> ```
> GOOS="darwin" go test -v -tags e2e ./utils/setup_test.go
> # runtime/cgo
> _cgo_export.c:3:10: fatal error: 'stdlib.h' file not found
> FAIL	command-line-arguments [build failed]
> FAIL
> ```

Solved by `export SDKROOT="$(xcrun --sdk macosx --show-sdk-path)"`
Do you like me to update the document in case other developers get caught?

**Comment by junekhan** [comment]
```
=== RUN   TestVerifyCommands
--- PASS: TestVerifyCommands (0.00s)
=== RUN   TestKubernetesConnection
--- PASS: TestKubernetesConnection (0.00s)
=== RUN   TestKubernetesVersion
    setup_test.go:38: kubernetes version: Client Version: v1.29.4
        Kustomize Version: v5.0.4-0.20230601165947-6ce0bf390ce3
        Server Version: v1.30.1
--- PASS: TestKubernetesVersion (1.12s)
=== RUN   TestSetupHelm
Password:
--- PASS: TestSetupHelm (11.51s)
=== RUN   TestSetupAzureManagedPrometheusComponents
    helper.go:548: Applying template: azureManagedPrometheusConfigMapTemplate
--- PASS: TestSetupAzureManagedPrometheusComponents (1.38s)
=== RUN   TestSetupCertManager
    setup_test.go:68: skipping cert manager is not required
--- SKIP: TestSetupCertManager (0.00s)
=== RUN   TestSetupWorkloadIdentityComponents
    setup_test.go:90: skipping as workload identity tests are disabled
--- SKIP: TestSetupWorkloadIdentityComponents (0.00s)
=== RUN   TestSetupAwsIdentityComponents
    setup_test.go:112: skipping aws identity tests are disabled
--- SKIP: TestSetupAwsIdentityComponents (0.00s)
=== RUN   TestSetupGcpIdentityComponents
    setup_test.go:134: skipping gcp identity tests are disabled
--- SKIP: TestSetupGcpIdentityComponents (0.00s)
=== RUN   TestSetupOpentelemetryComponents
    setup_test.go:173: skipping installing opentelemetry
--- SKIP: TestSetupOpentelemetryComponents (0.00s)
=== RUN   TestDeployKEDA
    helper.go:246: deleting namespace keda
    helper.go:299: waiting for namespace keda deletion
    helper.go:233: Creating namespace - keda
    setup_test.go:229: 
        	Error Trace:	/Users/june/Projects/gitroot/keda/tests/utils/setup_test.go:229
        	Error:      	Received unexpected error:
        	            	bash: /Users/june/Projects/gitroot/keda/bin/controller-gen: cannot execute binary file
        	            	make: *** [manifests] Error 126
        	Test:       	TestDeployKEDA
        	Messages:   	error deploying KEDA - bash: /Users/june/Projects/gitroot/keda/bin/controller-gen: cannot execute binary file
        	            	make: *** [manifests] Error 126
--- FAIL: TestDeployKEDA (2.47s)
=== RUN   TestVerifyKEDA
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-operator, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-operator, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-operator, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-operator, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-operator, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-operator, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-operator, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-operator, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-operator, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-operator, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-operator, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-operator, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-operator, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-operator, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-operator, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-operator, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-operator, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-operator, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-operator, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-operator, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-operator, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-operator, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-operator, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-operator, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-operator, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-operator, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-operator, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-operator, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-operator, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-operator, Current  - 0, Target - 1
    setup_test.go:240: 
        	Error Trace:	/Users/june/Projects/gitroot/keda/tests/utils/setup_test.go:240
        	Error:      	Should be true
        	Test:       	TestVerifyKEDA
        	Messages:   	replica count should be 1 after 3 minutes
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-metrics-apiserver, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-metrics-apiserver, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-metrics-apiserver, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-metrics-apiserver, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-metrics-apiserver, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-metrics-apiserver, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-metrics-apiserver, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-metrics-apiserver, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-metrics-apiserver, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-metrics-apiserver, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-metrics-apiserver, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-metrics-apiserver, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-metrics-apiserver, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-metrics-apiserver, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-metrics-apiserver, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-metrics-apiserver, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-metrics-apiserver, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-metrics-apiserver, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-metrics-apiserver, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-metrics-apiserver, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-metrics-apiserver, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-metrics-apiserver, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-metrics-apiserver, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-metrics-apiserver, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-metrics-apiserver, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-metrics-apiserver, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-metrics-apiserver, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-metrics-apiserver, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-metrics-apiserver, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-metrics-apiserver, Current  - 0, Target - 1
    setup_test.go:242: 
        	Error Trace:	/Users/june/Projects/gitroot/keda/tests/utils/setup_test.go:242
        	Error:      	Should be true
        	Test:       	TestVerifyKEDA
        	Messages:   	replica count should be 1 after 3 minutes
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-admission, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-admission, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-admission, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-admission, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-admission, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-admission, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-admission, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-admission, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-admission, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-admission, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-admission, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-admission, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-admission, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-admission, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-admission, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-admission, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-admission, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-admission, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-admission, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-admission, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-admission, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-admission, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-admission, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-admission, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-admission, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-admission, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-admission, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-admission, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-admission, Current  - 0, Target - 1
    helper.go:442: Waiting for deployment replicas to hit target. Deployment - keda-admission, Current  - 0, Target - 1
    setup_test.go:244: 
        	Error Trace:	/Users/june/Projects/gitroot/keda/tests/utils/setup_test.go:244
        	Error:      	Should be true
        	Test:       	TestVerifyKEDA
        	Messages:   	replica count should be 1 after 3 minutes
--- FAIL: TestVerifyKEDA (549.92s)
=== RUN   TestSetUpStrimzi
    setup_test.go:253: --- installing kafka operator ---
    helper.go:246: deleting namespace strimzi
    helper.go:299: waiting for namespace strimzi deletion
    helper.go:233: Creating namespace - strimzi
panic: test timed out after 10m0s
running tests:
	TestSetUpStrimzi (34s)

goroutine 191 [running]:
testing.(*M).startAlarm.func1()
	/opt/homebrew/Cellar/go/1.22.4/libexec/src/testing/testing.go:2366 +0x30c
created by time.goFunc
	/opt/homebrew/Cellar/go/1.22.4/libexec/src/time/sleep.go:177 +0x38

goroutine 1 [chan receive]:
testing.(*T).Run(0x140000db520, {0x105b12f3d?, 0x1400019db38?}, 0x1060bb7a8)
	/opt/homebrew/Cellar/go/1.22.4/libexec/src/testing/testing.go:1750 +0x32c
testing.runTests.func1(0x140000db520)
	/opt/homebrew/Cellar/go/1.22.4/libexec/src/testing/testing.go:2161 +0x40
testing.tRunner(0x140000db520, 0x1400019dc58)
	/opt/homebrew/Cellar/go/1.22.4/libexec/src/testing/testing.go:1689 +0xec
testing.runTests(0x1400000ee70, {0x106c9bd40, 0xd, 0xd}, {0x1400019dd18?, 0x104bf1ef0?, 0x106cb2900?})
	/opt/homebrew/Cellar/go/1.22.4/libexec/src/testing/testing.go:2159 +0x3b0
testing.(*M).Run(0x140005b43c0)
	/opt/homebrew/Cellar/go/1.22.4/libexec/src/testing/testing.go:2027 +0x5a4
main.main()
	_testmain.go:71 +0x16c

goroutine 188 [syscall]:
syscall.syscall6(0xc?, 0x14000051d50?, 0x1000110?, 0x1070c6720?, 0x90?, 0x106e1cf18?, 0x90?)
	/opt/homebrew/Cellar/go/1.22.4/libexec/src/runtime/sys_darwin.go:45 +0x68
syscall.wait4(0x14000051d18?, 0x104b9e0c8?, 0x90?, 0x106057320?)
	/opt/homebrew/Cellar/go/1.22.4/libexec/src/syscall/zsyscall_darwin_arm64.go:44 +0x4c
syscall.Wait4(0x1400027e0e0?, 0x14000051d54, 0x14000268160?, 0x106e1cf40?)
	/opt/homebrew/Cellar/go/1.22.4/libexec/src/syscall/syscall_bsd.go:144 +0x28
os.(*Process).wait(0x14000048600)
	/opt/homebrew/Cellar/go/1.22.4/libexec/src/os/exec_unix.go:43 +0x80
os.(*Process).Wait(...)
	/opt/homebrew/Cellar/go/1.22.4/libexec/src/os/exec.go:134
os/exec.(*Cmd).Wait(0x14000464000)
	/opt/homebrew/Cellar/go/1.22.4/libexec/src/os/exec/exec.go:897 +0x38
os/exec.(*Cmd).Run(0x14000464000)
	/opt/homebrew/Cellar/go/1.22.4/libexec/src/os/exec/exec.go:607 +0x38
os/exec.(*Cmd).Output(0x14000464000)
	/opt/homebrew/Cellar/go/1.22.4/libexec/src/os/exec/exec.go:991 +0xbc
github.com/kedacore/keda/v2/tests/helper.ExecuteCommand({0x14000460000?, 0x78?})
	/Users/june/Projects/gitroot/keda/tests/helper/helper.go:127 +0x24
command-line-arguments.TestSetUpStrimzi(0x140005f8000)
	/Users/june/Projects/gitroot/keda/tests/utils/setup_test.go:263 +0x220
testing.tRunner(0x140005f8000, 0x1060bb7a8)
	/opt/homebrew/Cellar/go/1.22.4/libexec/src/testing/testing.go:1689 +0xec
created by testing.(*T).Run in goroutine 1
	/opt/homebrew/Cellar/go/1.22.4/libexec/src/testing/testing.go:1742 +0x318

goroutine 114 [IO wait]:
internal/poll.runtime_pollWait(0x10ddaa858, 0x72)
	/opt/homebrew/Cellar/go/1.22.4/libexec/src/runtime/netpoll.go:345 +0xa0
internal/poll.(*pollDesc).wait(0x14000056b00?, 0x140006aa000?, 0x0)
	/opt/homebrew/Cellar/go/1.22.4/libexec/src/internal/poll/fd_poll_runtime.go:84 +0x28
internal/poll.(*pollDesc).waitRead(...)
	/opt/homebrew/Cellar/go/1.22.4/libexec/src/internal/poll/fd_poll_runtime.go:89
internal/poll.(*FD).Read(0x14000056b00, {0x140006aa000, 0x1300, 0x1300})
	/opt/homebrew/Cellar/go/1.22.4/libexec/src/internal/poll/fd_unix.go:164 +0x200
net.(*netFD).Read(0x14000056b00, {0x140006aa000?, 0x107172308?, 0x140004628d0?})
	/opt/homebrew/Cellar/go/1.22.4/libexec/src/net/fd_posix.go:55 +0x28
net.(*conn).Read(0x1400048e098, {0x140006aa000?, 0x140003898f8?, 0x104af0dfc?})
	/opt/homebrew/Cellar/go/1.22.4/libexec/src/net/net.go:179 +0x34
crypto/tls.(*atLeastReader).Read(0x140004628d0, {0x140006aa000?, 0x0?, 0x140004628d0?})
	/opt/homebrew/Cellar/go/1.22.4/libexec/src/crypto/tls/conn.go:806 +0x40
bytes.(*Buffer).ReadFrom(0x140001e9430, {0x1060c3820, 0x140004628d0})
	/opt/homebrew/Cellar/go/1.22.4/libexec/src/bytes/buffer.go:211 +0x90
crypto/tls.(*Conn).readFromUntil(0x140001e9188, {0x107175018, 0x1400048e098}, 0x140003899d0?)
	/opt/homebrew/Cellar/go/1.22.4/libexec/src/crypto/tls/conn.go:828 +0xd0
crypto/tls.(*Conn).readRecordOrCCS(0x140001e9188, 0x0)
	/opt/homebrew/Cellar/go/1.22.4/libexec/src/crypto/tls/conn.go:626 +0x35c
crypto/tls.(*Conn).readRecord(...)
	/opt/homebrew/Cellar/go/1.22.4/libexec/src/crypto/tls/conn.go:588
crypto/tls.(*Conn).Read(0x140001e9188, {0x140003bd000, 0x1000, 0x104b24f54?})
	/opt/homebrew/Cellar/go/1.22.4/libexec/src/crypto/tls/conn.go:1370 +0x168
bufio.(*Reader).Read(0x140003b6480, {0x140001b0ac0, 0x9, 0x104ae9208?})
	/opt/homebrew/Cellar/go/1.22.4/libexec/src/bufio/bufio.go:241 +0x1b4
io.ReadAtLeast({0x1060c2718, 0x140003b6480}, {0x140001b0ac0, 0x9, 0x9}, 0x9)
	/opt/homebrew/Cellar/go/1.22.4/libexec/src/io/io.go:335 +0xa0
io.ReadFull(...)
	/opt/homebrew/Cellar/go/1.22.4/libexec/src/io/io.go:354
golang.org/x/net/http2.readFrameHeader({0x140001b0ac0, 0x9, 0x105ebbe20?}, {0x1060c2718?, 0x140003b6480?})
	/Users/june/Projects/gitroot/keda/vendor/golang.org/x/net/http2/frame.go:237 +0x58
golang.org/x/net/http2.(*Framer).ReadFrame(0x140001b0a80)
	/Users/june/Projects/gitroot/keda/vendor/golang.org/x/net/http2/frame.go:498 +0x78
golang.org/x/net/http2.(*clientConnReadLoop).run(0x14000389f98)
	/Users/june/Projects/gitroot/keda/vendor/golang.org/x/net/http2/transport.go:2429 +0xcc
golang.org/x/net/http2.(*ClientConn).readLoop(0x140001b2300)
	/Users/june/Projects/gitroot/keda/vendor/golang.org/x/net/http2/transport.go:2325 +0x5c
created by golang.org/x/net/http2.(*ClientConn).goRun in goroutine 49
	/Users/june/Projects/gitroot/keda/vendor/golang.org/x/net/http2/transport.go:369 +0x38

goroutine 95 [IO wait]:
internal/poll.runtime_pollWait(0x10ddaaa48, 0x72)
	/opt/homebrew/Cellar/go/1.22.4/libexec/src/runtime/netpoll.go:345 +0xa0
internal/poll.(*pollDesc).wait(0x14000118420?, 0x14000470035?, 0x1)
	/opt/homebrew/Cellar/go/1.22.4/libexec/src/internal/poll/fd_poll_runtime.go:84 +0x28
internal/poll.(*pollDesc).waitRead(...)
	/opt/homebrew/Cellar/go/1.22.4/libexec/src/internal/poll/fd_poll_runtime.go:89
internal/poll.(*FD).Read(0x14000118420, {0x14000470035, 0x3cb, 0x3cb})
	/opt/homebrew/Cellar/go/1.22.4/libexec/src/internal/poll/fd_unix.go:164 +0x200
os.(*File).read(...)
	/opt/homebrew/Cellar/go/1.22.4/libexec/src/os/file_posix.go:29
os.(*File).Read(0x1400008a4a0, {0x14000470035?, 0x14000495d18?, 0x35?})
	/opt/homebrew/Cellar/go/1.22.4/libexec/src/os/file.go:118 +0x70
bytes.(*Buffer).ReadFrom(0x1400024adb0, {0x1060c2358, 0x1400020e0f0})
	/opt/homebrew/Cellar/go/1.22.4/libexec/src/bytes/buffer.go:211 +0x90
io.copyBuffer({0x1060c27d8, 0x1400024adb0}, {0x1060c2358, 0x1400020e0f0}, {0x0, 0x0, 0x0})
	/opt/homebrew/Cellar/go/1.22.4/libexec/src/io/io.go:415 +0x14c
io.Copy(...)
	/opt/homebrew/Cellar/go/1.22.4/libexec/src/io/io.go:388
os.genericWriteTo(0x4dfab4d35d8faf01?, {0x1060c27d8, 0x1400024adb0})
	/opt/homebrew/Cellar/go/1.22.4/libexec/src/os/file.go:269 +0x5c
os.(*File).WriteTo(0x106c85af0?, {0x1060c27d8?, 0x1400024adb0?})
	/opt/homebrew/Cellar/go/1.22.4/libexec/src/os/file.go:247 +0x60
io.copyBuffer({0x1060c27d8, 0x1400024adb0}, {0x1060c2418, 0x1400008a4a0}, {0x0, 0x0, 0x0})
	/opt/homebrew/Cellar/go/1.22.4/libexec/src/io/io.go:411 +0x98
io.Copy(...)
	/opt/homebrew/Cellar/go/1.22.4/libexec/src/io/io.go:388
os/exec.(*Cmd).writerDescriptor.func1()
	/opt/homebrew/Cellar/go/1.22.4/libexec/src/os/exec/exec.go:577 +0x44
os/exec.(*Cmd).Start.func2(0xde6ec38e337578a3?)
	/opt/homebrew/Cellar/go/1.22.4/libexec/src/os/exec/exec.go:724 +0x34
created by os/exec.(*Cmd).Start in goroutine 188
	/opt/homebrew/Cellar/go/1.22.4/libexec/src/os/exec/exec.go:723 +0x7cc

goroutine 96 [IO wait]:
internal/poll.runtime_pollWait(0x10ddaa950, 0x72)
	/opt/homebrew/Cellar/go/1.22.4/libexec/src/runtime/netpoll.go:345 +0xa0
internal/poll.(*pollDesc).wait(0x140001184e0?, 0x14000172000?, 0x1)
	/opt/homebrew/Cellar/go/1.22.4/libexec/src/internal/poll/fd_poll_runtime.go:84 +0x28
internal/poll.(*pollDesc).waitRead(...)
	/opt/homebrew/Cellar/go/1.22.4/libexec/src/internal/poll/fd_poll_runtime.go:89
internal/poll.(*FD).Read(0x140001184e0, {0x14000172000, 0x8000, 0x8000})
	/opt/homebrew/Cellar/go/1.22.4/libexec/src/internal/poll/fd_unix.go:164 +0x200
os.(*File).read(...)
	/opt/homebrew/Cellar/go/1.22.4/libexec/src/os/file_posix.go:29
os.(*File).Read(0x1400008a4b8, {0x14000172000?, 0xc2e0ba21115d1301?, 0x1dcbdad175b3ceb6?})
	/opt/homebrew/Cellar/go/1.22.4/libexec/src/os/file.go:118 +0x70
io.copyBuffer({0x1060c2858, 0x14000154000}, {0x1060c2358, 0x1400048e4f0}, {0x0, 0x0, 0x0})
	/opt/homebrew/Cellar/go/1.22.4/libexec/src/io/io.go:429 +0x18c
io.Copy(...)
	/opt/homebrew/Cellar/go/1.22.4/libexec/src/io/io.go:388
os.genericWriteTo(0x1?, {0x1060c2858, 0x14000154000})
	/opt/homebrew/Cellar/go/1.22.4/libexec/src/os/file.go:269 +0x5c
os.(*File).WriteTo(0x106c85af0?, {0x1060c2858?, 0x14000154000?})
	/opt/homebrew/Cellar/go/1.22.4/libexec/src/os/file.go:247 +0x60
io.copyBuffer({0x1060c2858, 0x14000154000}, {0x1060c2418, 0x1400008a4b8}, {0x0, 0x0, 0x0})
	/opt/homebrew/Cellar/go/1.22.4/libexec/src/io/io.go:411 +0x98
io.Copy(...)
	/opt/homebrew/Cellar/go/1.22.4/libexec/src/io/io.go:388
os/exec.(*Cmd).writerDescriptor.func1()
	/opt/homebrew/Cellar/go/1.22.4/libexec/src/os/exec/exec.go:577 +0x44
os/exec.(*Cmd).Start.func2(0xf6430d8d9376ca4?)
	/opt/homebrew/Cellar/go/1.22.4/libexec/src/os/exec/exec.go:724 +0x34
created by os/exec.(*Cmd).Start in goroutine 188
	/opt/homebrew/Cellar/go/1.22.4/libexec/src/os/exec/exec.go:723 +0x7cc
FAIL	command-line-arguments	600.336s
FAIL
```
Although it didn't pan out well in a while.

File: tests/internals/scaling_strategies/eager_scaling_strategy_test.go [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+var (
+	testNamespace        = fmt.Sprintf("%s-ns", testName)
+	rmqNamespace         = fmt.Sprintf("%s-rmq", testName)
+	scaledJobName        = fmt.Sprintf("%s-sj", testName)
+	queueName            = "hello"
+	user                 = fmt.Sprintf("%s-user", testName)
+	password             = fmt.Sprintf("%s-password", testName)
+	vhost                = "/"
+	connectionString     = fmt.Sprintf("amqp://%s:%s@rabbitmq.%s.svc.cluster.local/", user, password, rmqNamespace)
+	httpConnectionString = fmt.Sprintf("http://%s:%s@rabbitmq.%s.svc.cluster.local/", user, password, rmqNamespace)
```
[the remark this answers is not in the document]
Comment (junekhan) [reply tests/internals/scaling_strategies/eager_scaling_strategy_test.go:34]: Looks okay.

**Comment by junekhan** [comment]
Could you please rerun the checks? The [result](https://github.com/kedacore/keda/actions/runs/9708044090/job/26794194524?pr=5872#step:6:136) doesn't make sense.

**Comment by zroubalik** [comment]
@junekhan so sorry for the delay, I have been busy and missed the notification

**Pull request merged**